hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/Util.java

```diff
@@ -86,12 +92,34 @@ private Util() {
                     throw e;
                 }
             } catch (CommitFailureException e) {
+                failures++;
+
                 //if the backend fails the commit, we can retry
-                Log.LOGGER.debugf(e, "Commit attempt %d/%d failed: %s", failures, maxFailures, e.getMessage());
+                Log.LOGGER.debugf(e, "Commit attempt %d/%d failed. Will wait for %d ms before retrying." +
+                        " The failure message was: %s", failures, maxFailures, waitTime, e.getMessage());
 
                 lastException = e;
+
+                if (failures < maxFailures) {
+                    try {
+                        Thread.sleep(waitTime);
+                    } catch (InterruptedException ie) {
+                        Log.LOGGER.wInterruptedWhileWaitingForTransactionRetry();
+                        //reset the interruption flag
+                        Thread.currentThread().interrupt();
```

Comment:
Throwing the transaction failure afterwards is maybe a little bit problematic, but I think necessary, because that's the only possible way of telling the caller that we weren't able to commit successfully.